core/server/master/src/main/java/alluxio/master/meta/checkconf/ServerConfigurationChecker.java

```diff
@@ -0,0 +1,228 @@
+      return mConfigWarns;
+    }
+
+    /**
+     * @return the status of the configuration checker results
+     */
+    public Map<Scope, Status> getConfigStatus() {
+      return mConfigStatus;
+    }
+  }
+
+  /**
+   * Constructs a new {@link ServerConfigurationChecker}.
+   *
+   * @param masterRecord master configuration record
+   * @param workerRecord worker configuration record
+   */
+  public ServerConfigurationChecker(ServerConfigurationRecord masterRecord,
+      ServerConfigurationRecord workerRecord) {
+    mMasterRecord = masterRecord;
+    mWorkerRecord = workerRecord;
+    mConfigCheckReport = new ConfigCheckReport();
+  }
+
+  /**
+   * Checks the server-side configurations and records the check results.
+   */
+  public synchronized void regenerateReport() {
+    // Generate the configuration map from master and worker configuration records
+    Map<PropertyKey, Map<String, List<String>>> confMap = generateConfMap();
+
+    // Update the errors and warnings configuration
+    Map<Scope, List<WrongProperty>> confErrors = new HashMap<>();
+    Map<Scope, List<WrongProperty>> confWarns = new HashMap<>();
+
+    for (Map.Entry<PropertyKey, Map<String, List<String>>> entry : confMap.entrySet()) {
+      if (entry.getValue().size() >= 2) {
+        PropertyKey key = entry.getKey();
+        WrongProperty wrongProperty = new WrongProperty()
+            .setName(key.getName()).setValues(entry.getValue());
+        Scope scope = key.getScope().equals(Scope.ALL) ? Scope.SERVER : key.getScope();
+        if (entry.getKey().getConsistencyLevel().equals(ConsistencyCheckLevel.ENFORCE)) {
+          confErrors.putIfAbsent(scope, new ArrayList<>());
+          confErrors.get(scope).add(wrongProperty);
+        } else {
+          confWarns.putIfAbsent(scope, new ArrayList<>());
+          confWarns.get(scope).add(wrongProperty);
+        }
+      }
+    }
+
+    // Update configuration status
+    Map<Scope, Status> configStatus = new HashMap<>();
+    Status masterStatus = Status.PASSED;
+    Status workerStatus = Status.PASSED;
+
+    for (Map.Entry<Scope, List<WrongProperty>> entry : confWarns.entrySet()) {
+      if (entry.getValue().size() > 0) {
+        Scope scope = entry.getKey();
+        if (scope.contains(Scope.MASTER)) {
+          masterStatus = Status.WARN;
+        }
+        if (scope.contains(Scope.WORKER)) {
+          workerStatus = Status.WARN;
+        }
+      }
+    }
+    for (Map.Entry<Scope, List<WrongProperty>> entry : confErrors.entrySet()) {
+      if (entry.getValue().size() > 0) {
+        Scope scope = entry.getKey();
+        if (scope.contains(Scope.MASTER)) {
+          masterStatus = Status.FAILED;
+        }
+        if (scope.contains(Scope.WORKER)) {
+          workerStatus = Status.FAILED;
+        }
+      }
+    }
+
+    configStatus.put(Scope.MASTER, masterStatus);
+    configStatus.put(Scope.WORKER, workerStatus);
+    configStatus.put(Scope.SERVER,
```

Comment:
this is confusing because Scope.SERVER usually refers to properties that apply to both master and worker, but here we're using it for properties that apply to either master or worker. Instead of using Scope.SERVER, could we have a separate report field for the overall status?